File: 561.array-partition.java
/*
 * @lc app=leetcode id=561 lang=java
 *
 * [561] Array Partition
 */

// @lc code=start

import java.util.Arrays;

class Solution {
    public int arrayPairSum(int[] nums) {
        Arrays.sort(nums);
        int ret = 0;
        for (int ix = 0; ix != nums.length; ix += 2) {
            ret += nums[ix];
        }
        return ret;
    }
}
// @lc code=end

File: 561.array-partition.cpp
/*
 * @lc app=leetcode id=561 lang=cpp
 *
 * [561] Array Partition
 */

// @lc code=start

#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int arrayPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int sum = 0;
        for (auto iter = nums.begin(); iter < nums.end(); iter += 2) {
            sum += *iter;
        }
        return sum;
    }
};
// @lc code=end
